SystemBase updates

neveritt opened the pull request:
### Changes

Updated Control Toolbox to work with the new types in SystemsBase.
We should merge this after we merged the Place command.

neveritt committed:
updated according to updates in SystemsBase

aytekinar commented on src/c2d.jl:
```diff
@@ -42,31 +41,31 @@ end
   Dd    = D + C*M2
   x0map = [speye(nx) -M2]
-  StateSpace{Siso{T}}(Ad, Bd, Cd, Dd, Ts), x0map
+  ss(Ad, Bd, Cd, Dd, Ts), x0map
```
I will check this part. This was Cristian's implementation, enforcing some SISO/MIMO structure regardless of the type of `D` matrix. His version was correct. I'll fix this.

aytekinar commented on src/c2d.jl:
```diff
@@ -42,31 +41,31 @@ end
   α::T
   @compat function (::Type{GeneralizedBilinear}){T}(α::T = 0.5)
-    @assert α ≧ 0. && α ≦ 1. "GeneralizedBilinear: α must be between 0 and 1"
+    @assert α ≥ 0. && α ≤ 1. "GeneralizedBilinear: α must be between 0 and 1"
```
nice catch! stupid Atom autocompletion :(

aytekinar commented on src/c2d.jl:
```diff
@@ -42,31 +41,31 @@ end
   Dd    = D + α*(C*Bd)
   x0map = [speye(nx) zeros(nx, nu)]
-  StateSpace{Siso{T}}(Ad, Bd, Cd, Dd, Ts), x0map
+  ss(Ad, Bd, Cd, Dd, Ts), x0map
```
Same here. Maybe we can have some helper function for these conversions to avoid the enforcing inner constructor.

aytekinar commented on src/c2d.jl:
```diff
@@ -75,33 +74,33 @@ end
   Dd    = D
   x0map = [eye(nx) zeros(nx, nu)]
-  StateSpace{Siso{T}}(Ad, Bd, Cd, Dd, Ts), x0map
+  ss(Ad, Bd, Cd, Dd, Ts), x0map
```
again...

aytekinar commented on src/c2d.jl:
```diff
@@ -75,33 +74,33 @@ end
   Dd    = D + C*Bd
   x0map = [eye(nx) zeros(nx, nu)]
-  StateSpace{Siso{T}}(Ad, Bd, Cd, Dd, Ts), x0map
+  ss(Ad, Bd, Cd, Dd, Ts), x0map
```
same issue...

aytekinar reviewed (changes requested):
Some changes are wrong due to the enforcing type constructors used in `StateSpace`. I am on this!

Thanks for the rest :)

Commit:
Fix `c2d` for the new type system

Removed the use of inner construction `StateSpace{Val{:siso}}` for
`D::AbstractMatrix`.

neveritt committed:
Merge pull request #1 from aytekinar/5-base-patch

Fix `c2d` for the new type system
